### ISSUE TITLE
Problem: Inconsistent checks on "amount" values

### ISSUE THREAD

**Issue by ttmc** [bigchaindb/bigchaindb#2453]
The current JSON Schema file for transactions (`bigchaindb/common/schema/transaction_v2.0.yaml`) imposes the following restrictions on an output's "amount":
```yaml
  output:
    type: object
    additionalProperties: false
    required:
    - amount
    - condition
    - public_keys
    properties:
      amount:
        type: string
        pattern: "^[0-9]{1,20}$"
```
In other words, it allows anything from "0" to "99999999999999999999" (twenty nines).

We have _other code_ which imposes different, tighter, constraints on the value of output.amount. Specifically, in `bigchaindb/bigchaindb/common/transaction.py`, there is the following code:

```python
class Output(object):
    """An Output is used to lock an asset.
    Wraps around a Crypto-condition Condition.
        Attributes:
            fulfillment (:class:`cryptoconditions.Fulfillment`): A Fulfillment
                to extract a Condition from.
            public_keys (:obj:`list` of :obj:`str`, optional): A list of
                owners before a Transaction was confirmed.
    """

    MAX_AMOUNT = 9 * 10 ** 18

    def __init__(self, fulfillment, public_keys=None, amount=1):
        """Create an instance of a :class:`~.Output`.
            Args:
                fulfillment (:class:`cryptoconditions.Fulfillment`): A
                    Fulfillment to extract a Condition from.
                public_keys (:obj:`list` of :obj:`str`, optional): A list of
                    owners before a Transaction was confirmed.
                amount (int): The amount of Assets to be locked with this
                    Output.
            Raises:
                TypeError: if `public_keys` is not instance of `list`.
        """
        if not isinstance(public_keys, list) and public_keys is not None:
            raise TypeError('`public_keys` must be a list instance or None')
        if not isinstance(amount, int):
            raise TypeError('`amount` must be an int')
        if amount < 1:
            raise AmountError('`amount` must be greater than 0')
        if amount > self.MAX_AMOUNT:
            raise AmountError('`amount` must be <= %s' % self.MAX_AMOUNT)
        # ...
```
In other words, these constraints say output amount must be >= 1 and <= 9 * 10 ** 18

## Two Problems

1. The JSON Schema file is inconsistent with the checks in `common/transaction.py`. Either we should make them consistent or we should eliminate one set of checks, since there seems to be some redundancy.
1. Why are the checks (which are checks about transaction validity) happening inside the constructor (`__init__()`) code for `Output` objects? Shouldn't transaction construction and transaction validation be separate concerns?

**Comment by ttmc** [comment]
I've changed my mind about this issue. I think it's okay to think of the JSON Schema check as a cheap pre-filter that allows some "bad" output amount values through, such as "0", but such values later get caught by the code in `__init__()` for the Output class. While it's true that validation code should all live by itself and be isolated, that can be done as a separate/bigger task, as outlined in BEP-9.
